File: src/main/java/com/tort/mudai/RoomSnapshot.java
package com.tort.mudai;

import com.tort.mudai.mapper.Directions;

import java.util.Set;

public class RoomSnapshot {
    private String _locationTitle;
    private String[] _objectsPresent = new String[]{};
    private String[] _mobs = new String[]{};
    private String _locationDesc;
    private Set<Directions> _exits;

    public void setLocationTitle(String locationTitle) {
        _locationTitle = locationTitle;
    }

    public void setObjectsPresent(String[] objectsPresent) {
        _objectsPresent = objectsPresent;
    }

    public String getLocationTitle() {
        return _locationTitle;
    }

    public String[] getObjectsPresent() {
        return _objectsPresent;
    }

    public void setMobs(String[] _mobs) {
        this._mobs = _mobs;
    }

    public String[] getMobs() {
        return _mobs;
    }

    public void setLocationDesc(String locationDesc) {
        _locationDesc = locationDesc;
    }

    public String getLocationDesc() {
        return _locationDesc;
    }

    public Set<Directions> getExits() {
        return _exits;
    }

    public void setExits(Set<Directions> exits) {
        _exits = exits;
    }
}

File: src/main/scala/com/tort/mudai/RoomSnapshot.scala
package com.tort.mudai

import com.tort.mudai.mapper.{Item, Exit}
import scalaz.@@
import com.tort.mudai.mapper.Mob.FullName


case class RoomSnapshot(
                         title: String,
                         desc: String,
                         exits: Set[Exit],
                         objectsPresent: Seq[ItemAndNumber] = Seq(),
                         mobs: Seq[String] = Seq()
                         ) extends RoomKey

object RoomSnapshot {
  def apply(key: RoomKey): RoomSnapshot = RoomSnapshot(key.title, key.desc, key.exits)
}

case class ItemAndNumber(item: String @@ Item.FullName, number: Int)

trait RoomKey {
  def title: String

  def desc: String

  def exits: Set[Exit]
}
